### PR TITLE
Build CUB and Thrust tests with assertions

### PR DESCRIPTION
**Pull request by bernhardmgruber**

### CODE REVIEW

**Review by miscco — APPROVED** [review]

**Review by miscco — APPROVED** [review]

File: cub/cub/detail/fast_modulo_division.cuh [hunk 1]
```diff
@@ -109,8 +109,11 @@ multiply_extract_higher_bits(T value, R multiplier)
 {
   static_assert(supported_integral<T>::value, "unsupported type");
   static_assert(supported_integral<R>::value, "unsupported type");
+  _CCCL_DIAG_PUSH
+  _CCCL_DIAG_SUPPRESS_ICC(186) // pointless comparison of unsigned integer with zero
```
Comment (miscco) [inline cub/cub/detail/fast_modulo_division.cuh:113]: Do we want to add an `_CCCL_IF_CONSTEXPR(_CCCL_TRAIT(is_signed, T))`

Comment (bernhardmgruber) [reply cub/cub/detail/fast_modulo_division.cuh:113]: I don't entrust ICC with the competency to not issue the warning if we put a runtime if around the assert before C++17. Plus, ICC is going away soon anyway. I would like to keep the code as is.

Comment (miscco) [reply cub/cub/detail/fast_modulo_division.cuh:113]: Its less about whether ICC is able to suppress the warning, but whether we want pay the price of the check in debug mode for unsigned integers

Comment (bernhardmgruber) [reply cub/cub/detail/fast_modulo_division.cuh:113]: Ah sorry. I thought your comment was related to my warning suppression. Follow-up: #2997

**Pull request merged**